content/blog/warnings.md

```diff
@@ -0,0 +1,130 @@
+
+```
+local({
+  for (i in 1:50) {
+    Sys.timezone(FALSE)
+  }
+  readRDS("nosuchfile")
+})
+#> Error in gzfile(file, "rb") : cannot open the connection
+#> In addition: There were 50 or more warnings (use warnings() to see the first 50)
+```
+
+Similar to above, except we do not get the real error here anymore. Worse, when following the instructions and running `warnings()` our true error is not found as every warning printed is about a defunct argument to `Sys.timezone`
+
+```
+#> 1: Sys.timezone(location = FALSE) is defunct and ignored
+#> 2: Sys.timezone(location = FALSE) is defunct and ignored
+...
+#> 50: Sys.timezone(location = FALSE) is defunct and ignored
+```
+
+This means that the catastrophic warning produced by `readRDS` is swamped by the informational one by `Sys.timezone` and the user has limited information with which to start debugging.
+
+The above example is obviously contrived but it's not that hard to accumulate 50 warnings - many packages are quite chatty with warnings as many packages use them for informational messages. Alternatively, you might have `options(warnPartialMatchDollar = TRUE)` set so that partial matching produces warnings and have used code like
+
+```
+seq(1, 10, length = 11)
+#>  [1]  1.0  1.9  2.8  3.7  4.6  5.5  6.4  7.3  8.2  9.1 10.0
+#> Warning message:
+#> In seq.default(1, 10, length = 11) :
+#>   partial argument match of 'length' to 'length.out'
+```
+
+There are really very many warnings produced by base R that should be considered as errors, and they are too easy to ignore.  Things like
+
+```
+y <- Map(`+`, 1:5, 1:4)
+#> Warning message:
+#> In mapply(FUN = f, ..., SIMPLIFY = FALSE) :
+#>   longer argument not a multiple of length of shorter
+```
+
+are almost always a bug and should be verified and fixed.  The use of warning here makes this job harder for you as a programmer - try not to impose this same cost on your users.
+
+## When are warnings OK?
+
+The most common case where I feel that warnings are absolutely the right choice is deprecating features.  If your package will soon stop supporting a feature (removing an argument, changing a data format), the program _can_ continue, and you want the user to update the call, then warnings provide a good way of conveying that.
+
+```
+package::function(a = 1)
+#> Warning message:
+#> In f(1) : Use of argument 'a' is deprecated, please use 'b'
+```
+
+this approach has been formalised in both base R (`?.Deprecated`) and within the tidyverse's ["lifecycle" package](https://cran.r-project.org/web/packages/lifecycle/vignettes/lifecycle.html#deprecating-arguments)
+
+## TL;DR
+
+In my opinion, most of the time people use `warning` it would be better as `message` or `stop`. Don't display a warning and continue if there's a good chance that the program will not run correctly and don't bother the user with a warning if it's just informational.
```

Comment:
Maybe useful to add a note about strategy for dealing with warnings from other packages, which you touch on above. Fix them as it may lead to bugs further down the line or if not at least understand the reason for the warning and convince yourself that it shouldn't be an error? Is there ever value to suppressing warnings of a particular type you know can be ignored if they are difficult to resolve?

Reply:
Thanks for the suggestion - I added a section with an example that has caused some pain in the past